crates/rooch/src/commands/move_cli/commands/run_function.rs

```diff
@@ -147,10 +158,16 @@ impl CommandAction<ExecuteTransactionResponseView> for RunFunction {
 
                     context
                         .sign_and_execute(sender, action, Some(password), max_gas_amount)
-                        .await
+                        .await?
                 }
             }
+        };
+
+        if dry_run_result.raw_output.status != KeptVMStatusView::Executed {
+            result.error_info = Some(dry_run_result);
         }
```

Comment:
Put this check before `execute(tx)`, or skip execute(tx) if dry_run_result is not KeptVMStatusView::Executed. 

If the dry run result is not Executed, we should not try to execute it.

The publish command also has this mistake. cc @steelgeek091

Reply:
> Put this check before `execute(tx)`, or skip execute(tx) if dry_run_result is not KeptVMStatusView::Executed.
> 
> If the dry run result is not Executed, we should not try to execute it.
> 
> The publish command also has this mistake. cc @steelgeek091

Shall I fix them together?